### DISCUSSION TITLE
Build OpenRadioss using Intel Compiler

### DISCUSSION THREAD

**Discussion by andy4431** [OpenRadioss/OpenRadioss#577] [Q&A] [answered]
Hi,
Has anyone tried building OpenRadioss using the Intel compiler with the Intel MPI?  

My reason is to see if the solver will be faster as compared to using the GNU Compiler with OpenMPI.

Thanks,
Andy

**Comment by elequiniou** [comment]
Hi Andy, good point. For now, we focused on open-source tools with GNU compiler and OpenMPI but use of latest Intel OneAPI is under investigation, stay tuned!

**Reply by ehan08** [reply]
Hi Eric,

Would you like to provide the compiler flags for the Intel Fortran compiler? Thanks.

**Reply by elequiniou** [reply]
Sure, we are working on it. In the mean time, we are transitioning from some older version of the Intel compiler so-called "classic" to the LLVM ones available with the latest OneAPI and which can be downloaded free of charge from Intel. It takes some time to fix few minor porting issues and review optimal compiler flags, so apologies for this delay.

**Reply by nizo303** [reply]
Great to hear! I was just looking into compiling using the Fortran compiler in OneAPI. Looking forward to hearing more about your efforts. Keep up the good work!

**Reply by rashinde45** [reply]
Hey is OpenRadioss is compatible to built with Clang compiler ?

**Reply by OlivierWienholtz** [reply]
Hello,
There are for ifx but not clang.
to use with ifx, use ./build_script.sh -arch=linux4_ifx

**Reply by elequiniou** [reply]
Hi, well it is more FLang has 95% of the source code is Fortran, but yes it is possible to use LLVM based compilers. Thanks.

**Comment by panda1100** [comment, marked as answer]
Hi @andy4431 

I successfully built OpenRadioss using Intel oneAPI  based on @laurent-altr 's work  
https://github.com/OpenRadioss/OpenRadioss/commit/ae0dc9248f999a81a2f82e54d2f4591ee121db53, https://github.com/OpenRadioss/OpenRadioss/commit/431a25f18405e35efcec0a5d97a471e5affb6536 and my PR https://github.com/OpenRadioss/OpenRadioss/pull/759

**Reply by andy4431** [reply]
Hi @panda1100,
I'm glad to hear it works with the Intel oneAPI.  

Did you happen to compare run times between the binaries from Intel oneAPI and intel MPI versus the GNU compiler with OpenMPI?

Thanks,
Andy

**Reply by panda1100** [reply]
@andy4431 No performance comparison but we wrote how-to article https://ciq.co/blog/intel-oneapi-build-of-openradioss/
